llama.cpp

```diff
@@ -5105,13 +5251,15 @@ static bool llm_load_tensors(
     // load tensor data
     for (auto & it : ctx_bufs) {
         ggml_context * ctx = it.first;
-        ggml_backend_buffer_t buf = it.second;
-        if (!ml.load_all_data(ctx, progress_callback, progress_callback_user_data, buf, use_mlock ? &model.mlock_mmap : NULL)) {
+        std::vector<ggml_backend_buffer_t> bufs = it.second;
```

Comment:
Avoid copying the entire vector.

```suggestion
        auto & bufs = it.second;
```